### PR TITLE
move build configuration into `pyproject.toml`

### PR DESCRIPTION
**Pull request by zacharyburnett**
`setuptools` now supports the `[project]` table, which is defined by [PEP621](https://peps.python.org/pep-0621).

Additionally, `setuptools` now supports its own entry in `pyproject.toml` called `[tool.setuptools]` (https://github.com/pypa/setuptools/issues/1688, https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration); however, it comes with the following disclaimer:
> Support for declaring configurations not standardized by [PEP 621](https://peps.python.org/pep-0621/) (i.e. the `[tool.setuptools]` table), is still in **beta** stage and might change in future releases.

It is likely that this functionality with `setuptools` will leave beta stage upon the release of Python 3.11, with its native implementation of `tomli`

Given this, we can attempt to consolidate the build configuration into a single `pyproject.toml` file that can possibly be read by other build systems in the future.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Topic :: Scientific/Engineering :: Astronomy",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "astropy>=4.1",
```
Comment (nden) [inline pyproject.toml:17]:
```suggestion
    "astropy>=5.1",
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+[tool.distutils.bdist_wheel]
+universal = 1
+
+[tool.pytest.ini_options]
+minversion = "3"
+norecursedirs = [
+    ".eggs",
+    "build",
+    "docs/_build",
+    "relic",
```
Comment (nden) [inline pyproject.toml:67]: It'd be good to stop using relic in stistools. It is an old in-house build system managing releases

**Review by nden — CHANGES_REQUESTED** [review]

**Comment by zacharyburnett** [comment]
removing reliance on `relic` would also likely solve this issue: https://github.com/spacetelescope/stenv/issues/78

**Review by nden — APPROVED** [review]

**Comment by sean-lockwood** [comment]
@zacharyburnett -
The RTD site was rebuilt around the time this PR was merged.  It looks like the docs built without `autodoc`, as most of the function descriptions are now blank.

E.g.:
https://stistools.readthedocs.io/en/latest/inttag.html

image

**Comment by zacharyburnett** [comment]
> The RTD site was rebuilt around the time this PR was merged. It looks like the docs built without autodoc, as most of the function descriptions are now blank.

That is strange; they populate on HTML built locally on my machine. I will look into this issue.

**Pull request merged**